cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java

```diff
@@ -490,6 +549,8 @@ public void delete(StorageLocation storageLocation)
     public SortedSet<StorageMetadata> list(String storageBucket)
         throws StorageEngageException, TransientException {
         StringBuilder path = new StringBuilder();
+        path.append(CONTENT_FOLDER);
```

Comment:
Use contentPath (after rename) directly here as the parent. Traversal through subdirectories should be always be done via 
```Path p = contentPath.resolve(sbpath);```

Reply:
believe I've got this... the storage bucket part of the path is still built in a string builder (particularly the URIBUCKET case,) - but then that string is added via .resolve to the contentPath.